### PR TITLE
Add a feature flag for a failure because of an heartbeat failed check

### PR DESCRIPTION
**Pull request by benmoriceau**
## What
Add a feature flag that will then be used to protect jobs from failing because of a missed heartbeat.

### CODE REVIEW

File: configs/flags.yml [hunk 1]
```diff
@@ -1,3 +1,5 @@
 flags:
   - name: performance.backgroundJsonSchemaValidation
     enabled: true
+  - name: heartbeat.fail_sync
```
Comment (colesnodgrass) [inline configs/flags.yml:4]: The docs are up yet, but the existing feature-flags are all camelCased.  Could we camelCase this one for consistency?

**Review by colesnodgrass — APPROVED** [review]
- camelCase the flag for consistency with other flags
- create this flag in LD (if you haven't yet)

File: configs/flags.yml (hunk 1 above)
Comment (benmoriceau) [reply configs/flags.yml]: Done

**Review by jdpgrailsdev — APPROVED** [review]
:shipit:

**Comment by colesnodgrass** [comment]
@benmoriceau heads-up that the `flags.yml` file was moved to the root directory in #22158

**Comment by benmoriceau** [comment]
@colesnodgrass could you review the flag in LD? https://app.launchdarkly.com/default/production/features/heartbeat.failSync/targeting It is the first one, I'm creating and I would like to make sure that I didn't do a mistake.

**Pull request merged**